test/extension/handlers/lifecycle/handlers.go

```diff
@@ -126,6 +126,23 @@ func (h *Handler) DoAfterClusterUpgrade(ctx context.Context, request *runtimehoo
 	}
 }
 
+// DoBeforeClusterDelete implements the BeforeClusterDelete hook.
+func (h *Handler) DoBeforeClusterDelete(ctx context.Context, request *runtimehooksv1.BeforeClusterDeleteRequest, response *runtimehooksv1.BeforeClusterDeleteResponse) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("BeforeClusterDelete is called")
+	cluster := request.Cluster
+
+	if err := h.readResponseFromConfigMap(ctx, cluster.Name, cluster.Namespace, runtimehooksv1.BeforeClusterDelete, response); err != nil {
+		response.Status = runtimehooksv1.ResponseStatusFailure
+		response.Message = err.Error()
+		return
+	}
+	if err := h.recordCallInConfigMap(ctx, cluster.Name, cluster.Namespace, runtimehooksv1.BeforeClusterDelete, response); err != nil {
```

Comment:
This will not be blocking for this PR but perhaps we should record the call in a `defer` block so that we record the call even if we wish to send back a failure response.

Same with other handlers.

Reply:
I guess this depends on what we want to record. If it's only successful responses then the current implementation is correct. If we also want to record response-lookup errors then doing is in defer or maybe easier, just not returning in l.138 would be correct.

I'm not sure how much it matters given that we only use it in our e2e test

Reply:
I'm going to change this in #6817 so let's discuss improvements to how it's done over there.